### ISSUE TITLE
stacks-node crashed

### ISSUE THREAD

**Issue by MaksimalistT** [stacks-network/stacks-blockchain#2970]
**Describe the bug**
stacks-node process crashed
works after restart

**Steps To Reproduce**
no info

**Environment (please complete the following information):**
- OS: Amazon Linux
- Rust version: rustc 1.49.0 (e1884a8e3 2020-12-29)
- Stacks-node 2.05.0.0.0

**Additional context**
[debug_with_crash_tail.log](https://github.com/blockstack/stacks-blockchain/files/7698481/debug_with_crash_tail.log)

**Comment by jcnelson** [comment]
Thanks for reporting!  The full stack trace was:

```
Process abort due to thread panic: panicked at 'ERROR: Failed to begin new unconfirmed MARF block for dd756178192819bb98b1948e976f5f89e92c98a8ee5f4cfcdb870aa9bfe65819): InProgressError', src/clarity_vm/database/marf.rs:199:39
   0: stacks_node::main::{{closure}}
             at /src/testnet/stacks-node/src/main.rs:58
   1: std::panicking::rust_panic_with_hook
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:628
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:521
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:141
   4: rust_begin_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:517
   5: core::panicking::panic_fmt
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:101
   6: core::result::unwrap_failed
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/result.rs:1617
   7: core::result::Result<T,E>::expect
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/result.rs:1259
      blockstack_lib::clarity_vm::database::marf::MarfedKV::begin_unconfirmed
             at /src/src/clarity_vm/database/marf.rs:199
   8: blockstack_lib::clarity_vm::clarity::ClarityInstance::begin_unconfirmed
             at /src/src/clarity_vm/clarity.rs:456
   9: blockstack_lib::chainstate::stacks::db::StacksChainState::chainstate_begin_unconfirmed
             at /src/src/chainstate/stacks/db/mod.rs:1863
      blockstack_lib::chainstate::stacks::db::StacksChainState::begin_unconfirmed
             at /src/src/chainstate/stacks/db/mod.rs:1882
  10: blockstack_lib::chainstate::stacks::miner::StacksMicroblockBuilder::resume_unconfirmed
             at /src/src/chainstate/stacks/miner.rs:248
  11: stacks_node::neon_node::mine_one_microblock
             at /src/testnet/stacks-node/src/neon_node.rs:350
      stacks_node::neon_node::try_mine_microblock
             at /src/testnet/stacks-node/src/neon_node.rs:474
      stacks_node::neon_node::run_microblock_tenure
             at /src/testnet/stacks-node/src/neon_node.rs:522
  12: stacks_node::neon_node::spawn_miner_relayer::{{closure}}
             at /src/testnet/stacks-node/src/neon_node.rs:1117
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:125
  13: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:481
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panic/unwind_safe.rs:271
      std::panicking::try::do_call
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:403
      std::panicking::try
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:367
      std::panic::catch_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panic.rs:129
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:480
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:227
  14: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys/unix/thread.rs:106
  15: start_thread
  16: __GI___clone
```

**Comment by jcnelson** [comment]
This was fixed by #3016; going to go ahead and close.  The fix will be in the next stable release.
